app/src/main/java/com/example/demomvvm/MainViewModel.kt

```diff
@@ -0,0 +1,27 @@
+package com.example.demomvvm
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.demomvvm.data.MovieDataSource
+import com.example.demomvvm.data.model.Movie
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class MainViewModel(
+    private val remote: MovieDataSource
+) : ViewModel() {
+
+    var movies: MutableLiveData<Movie> = MutableLiveData()
```

Comment:
If a class has two properties which are conceptually the same but one is part of a public API and another is an implementation detail, use an underscore as the prefix for the name of the private property:

``` Kotlin
class C {
    private val _elementList = mutableListOf<Element>()

    val elementList: List<Element>
         get() = _elementList
}
```
https://kotlinlang.org/docs/coding-conventions.html#names-for-backing-properties
https://developer.android.com/topic/libraries/architecture/viewmodel

Reply:
Em đã tìm hiểu sự khác nhau giữa LiveData và MutableLiveData chưa ?

Reply:
LiveData thì ko sửa đổi dữ liệu đc
MutableLiveData thì có thể ạ